syntax = "proto2";

message Empty2 {}

message WeightQuery {
  optional uint32 version = 1;
}

message Weights {
  enum Status {
    OK = 1;
    UP_TO_DATE = 2;
  }
  optional Status status = 1;
  optional int32 version = 2;
  optional bytes data = 3;
  optional string log_dir = 4;
}


service ModelService {
  rpc PutWeights (Weights) returns (Empty2) {}
  rpc GetWeights (WeightQuery) returns (Weights) {}
}
